### ISSUE TITLE
[Bug]: got "connection reset" when using HTTPS connect locally deploy minIO behind reverse proxy

### ISSUE THREAD

**Issue by 77-arthur** [remotely-save/remotely-save#509]
### What happened?

- plugin version：0.3.40
- obsidian version：1.5.3-public
- MinIO version：RELEASE.2022-10-15T19-57-03Z
- reverse proxy：nginx
- browser：Firefox、chrome
- public IP：yes

## Environment:
I deployed MinIO, reverse proxy on a local server with a public IP and resolved domain name.

## Problem Description:
When I test connecting to MinIO via HTTPS in Obsidian on Win11 it prompts "connection reset".

## Additional instructions:
I can connect to MinIO and upload files via local IP in Obsidian on Win11.
Beyond that, browsers and [mc](https://min.io/docs/minio/linux/reference/minio-mc.html) can connect and operate MinIO via local IP or HTTPS.

# 中文说明

## 环境：
我在一台具有公共IP并解析域名的本地服务器上部署了MinIO、反向代理。

## 问题描述：
当我在Win11的Obsidian中通过HTTPS测试连接MinIO时提示“connection reset”。

## 补充说明：
我在Win11的Obsidian中可以通过本地IP连接MinIO和上传文件。
除此之外，浏览器和[mc](https://minio.org.cn/docs/minio/linux/reference/minio-mc.html)可以通过本地IP或HTTPS连接和操作MinIO。

### What OS are you using?

Windows

### What remote cloud services are you using? (Please choose the specified one if it's in the list)

S3 (MinIO)

### Version of the plugin

0.3.40

### Version of Obsidian

1.5.3

### Using password or not

- [ ] Yes.

### Ensure no sensitive information

- [X] I ensure that no sensitive information is submitted in the issue.

**Comment by fyears** [comment]
any proxy do you use on browser?

**Comment by 77-arthur** [comment]
yes I'm running clash(Rule mode) on Win11. But the domian use Direct.

![图片](https://github.com/remotely-save/remotely-save/assets/50293747/861e2847-dcd0-4885-9c82-993ba63fa580)

**Comment by fyears** [comment]
试试本地也关掉代理，用真的直连，看浏览器等能不能连上？这个错误显然就是网络中断的问题了。估计我这边也没啥办法修理

**Comment by 77-arthur** [comment]
> 试试本地也关掉代理，用真的直连，看浏览器等能不能连上？这个错误显然就是网络中断的问题了。估计我这边也没啥办法修理

我关闭代理后，浏览器和之前一样域名和IP都正常。不过插件里测试域名，提示变成“name not resolved"不能解析域名。这有点奇怪，我在Win11里设置了静态DNS。感觉obsidian的网络配置有点问题，但下一步不知道改哪里了

**Comment by 77-arthur** [comment]
除此之外，我刚刚用IP连接上了MinIO（设置里能通过连接测试），但是我试着点击了左侧栏的手动同步，也在报错”request failed status 404“。想请问这里的404是指仓库文件404还是MinIO里404呢？Obsidian仓库在我用webdav映射的盘上，能正常修改和查看文件。

**Comment by fyears** [comment]
obsidian （和插件本身）应该没啥特殊设置，考虑一下查看一下网络设置，或者用全局代理试试吧，这里肯定是网络问题，修理不了

**Comment by fyears** [comment]
> 想请问这里的404是指仓库文件404还是MinIO里404呢？

是请求 minio 拉数据（文件列表）时候404，是不是 path style 有问题或者权限有问题？

试试对着这个走，我用官方网站和本地 127.0.0.1 的都成功了的：https://github.com/remotely-save/remotely-save/blob/master/docs/remote_services/s3_minio/README.md

**Comment by 77-arthur** [comment]
根据引导将URL Style设置为path style后能通过域名连接MinIO。

不过我发现了可以稳定复现404的操作

- 根据引导创建一个空桶 `test`，不修改配置。
- 初次手动上传文件到`test`桶（成功）
- 初次成功后，再次手动上传便会一直404
- 试过IP和https域名都会404

![图片](https://github.com/remotely-save/remotely-save/assets/50293747/bb37bfe8-1585-4c34-9598-f9a3e1314e4f)

这里是桶策略问题吗？我试着配置为public也依然存在404

**Comment by fyears** [comment]
桶叫做test，文件夹名也叫做test，文件名叫做 test file.md？

你的 minio 是什么版本？minio 宿主机是什么操作系统？在 minio 网页版可以看到 key 为 `test/` （注意有后缀斜杠）的 object吗？

同样的问题，可以用本地 windows 电脑，直接用 minio.exe 复现吗？我是直接下载了 minio.exe 在 windows 开一个单机版本测试的

**Comment by 77-arthur** [comment]
宿主机OS：unRaid 6.9.2
以docker容器运行MinIO
MinIO version：RELEASE.2022-10-15T19-57-03Z
因为MinIO 2022-10-29的版本中开始使用`SNSD` 模式，unRaid不能以docker直接运行MinIO，所以停在更新前一个版本。

> 桶叫做test，文件夹名也叫做test，文件名叫做 test file.md？

抱歉，下午测试时没注意到桶名和文件夹名重复，确实会有点让人困惑。后续测试中，我会注意命名问题。

桶名：`test`
文件夹名：`test`
文件名：`test file.md`

> 在 minio 网页版可以看到 key 为 `test/` （注意有后缀斜杠）的 object吗？

后台中可以看见第一次上传成功的文件

![图片](https://github.com/remotely-save/remotely-save/assets/50293747/a8916e75-159e-4fcc-8588-9053c6791e06)


> 同样的问题，可以用本地 windows 电脑，直接用 minio.exe 复现吗？

我并未尝试过在Windows本地运行MinIO，目前只测试了通过局域网IP和反代https域名连接局域网服务器中的MinIO。我尽量在明晚完成测试给您回复。

**Comment by fyears** [comment]
这里看起来是传了文件，不是文件夹
估计是这个问题

**Comment by fyears** [comment]
你也可以直接用 mc 创建一个 `abc/` 为 key 的 0byte 文件，看是否成功

**Comment by 77-arthur** [comment]
同时测试了https和ip都能创建和删除`abc/`文件夹

![图片](https://github.com/remotely-save/remotely-save/assets/50293747/d1a6040a-6596-4feb-b4a2-e43bc141cd92)

**Comment by 77-arthur** [comment]
> 这里看起来是传了文件，不是文件夹
> 估计是这个问题

但之前的debug记录显示是先成功创建了`test`文件夹，再上传的`test file.md`

`test file.md`也在`test`文件夹下的

**Comment by 77-arthur** [comment]
我编译了和docker版本相同的minio Windows `exe`文件，并通过了本地测试。

最终我在备份后重新部署了MinIO，这一次没有再出现之前只能上传一遍的问题。

感谢您这几天的积极回复！

**Comment by fyears** [comment]
好的，估计就是版本问题
